### PR TITLE
as a end user I would like to be able to attach preset memes to my status update on gitcoin, so i can share memes

### LINKED ISSUE #5882: as a end user I would like to be able to attach preset memes to my status update on gitcoin, so i can share memes
### User Story

[comment]: # (As a <user type>, I want to <task> so that <goal>.)
as a end user I would like to be able to attach preset memes to my status update on gitcoin, so i can share memes

### Why Is this Needed

[comment]: # (Describe the problem and why this task is needed. Provide description of the current state, what you would like to happen, and what actually happen)
*Summary*:
people are visual

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*:
feature

### Current Behavior
[comment]: # (Describe what actually happened.)
no image upload

### Expected Behavior
[comment]: # (Describe what you expected to happen.)
1. build on top of https://github.com/gitcoinco/web/issues/5832
2. add the ability to search gifs from giphy by keyword
3. user can select a gif from the list, and if they seelct it itll be attached to the status update
4. gif will be shown on the status update.

example of how facebook does this
Screen Shot 2020-01-24 at 2 48 41 PM

### Definition of Done
[comment]: # (Any other information that would be useful, bullets are helpful.)
see above

### Additional Information
[comment]: # (Any other information that would be useful, content, screenshots, etc.)

### PR DESCRIPTION
**Pull request by zoek1**
##### Description
- [X] build on top of #5832 - I'm based on  #5880 due to merge conflicts
- [X] add the ability to search gifs from giphy by keyword
Home _ Gitcoin (1)
- [X] user can select a gif from the list, and if they select it, it will be attached to the status update
Home _ Gitcoin (2)
- [X] gif will be shown on the status update.
(1) Home _ Gitcoin

I've validated generate no conflict with links and youtube videos.

##### Refers/Fixes
#5882

### CODE REVIEW

**Comment by zoek1** [comment]
It's ready to review! :grinning:

**Comment by danlipert** [comment]
Hey @zoek1 - I know you've got a lot of cool PRs up, but you've got to clean up the commit history on these. They are hard to review when there are commits from so many different features in each PR. Let me know if that doesn't make sense to you.

**Comment by zoek1** [comment]
Definitely is hard when that happens, what can I do to reduce this?  I could redo the history and split the PR @danlipert

**Comment by octavioamu** [comment]
Looking nice. Just an important performance point. Can we use the giphy video version instead of the .gif? Most social networks do that because gif are fat.

**Comment by zoek1** [comment]
Sure, no problem!

**Comment by danlipert** [comment]
@zoek1 just rebase the branch to remove the other commits, or cherry-pick the relevant commits onto a new branch. And make sure to always branch off latest master

**Comment by owocki** [comment]
@zoek1 @ me when u put in ur fix pls!

**Comment by zoek1** [comment]
Sure! I'll notify you here

**Comment by zoek1** [comment]
@owocki I fixed the commits, due this PR is based on #5880, I recommend merge that one first to clear the history in this PR. So the order is: 
1. Merge #5889 
2. Merge this PR #5898 
3. Merge #5884

**Comment by zoek1** [comment]
@octavioamu I used webp instead of mp4 version. I tested the webp vs mp4, it results that mp4 performs better at a fixed size but if we want maintain the original proportion (size) of the gif, the webp performs much better.
I can open another PR if you decide is better to go with a fixed size and use mp4 format. :smile:

**Comment by zoek1** [comment]
@owocki ready to review!

**Comment by owocki** [comment]
thanks @zoek1 ; the plan is to sprint to get this in by weds deploy i think.   allso checkout the new `stable` branch and the new experience at gitcoin.co -- well likely have to adapt this to that design (can tip if it creates a lot more scope creep)

**Comment by zoek1** [comment]
Ok, today I'll check the stable branch to adapt this!

File: app/assets/v2/js/status.js [hunk 1]
```diff
@@ -1,5 +1,45 @@
+const url_re = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,10}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/;
+const youtube_re = /(?:https?:\/\/|\/\/)?(?:www\.|m\.)?(?:youtu\.be\/|youtube\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))([\w-]{11})(?![\w-])/;
+const giphy_re = /(?:https?:\/\/)?(?:media0\.)?(?:giphy\.com\/media\/)/;
+
 $(document).ready(function() {
+  var embedded_resource = '';
+  const GIPHY_API_KEY = '1WfeOI2i4lJiBKvO2Q1W3yUqdjQ27UTy';
```
Comment (octavioamu) [inline app/assets/v2/js/status.js]: can we move this to settings.py?

File: app/townsquare/templates/townsquare/index.html [hunk 2]
```diff
@@ -161,6 +161,7 @@ <h4 class="pt-3">{{type|title}} Action</h4>
 {% endblock %}
 
 {% block scripts %}
+<script src="{% static "v2/js/lib/yall.min.js" %}"></script>
```
Comment (octavioamu) [inline app/townsquare/templates/townsquare/index.html]: Can we try with native lazy loading instead? I made this with kudos and is working really nice, also with a fallback using observe
https://github.com/gitcoinco/web/pull/5940

File: app/assets/v2/js/status.js (hunk 1 above)
Comment (zoek1) [reply app/assets/v2/js/status.js]: Sure, no problem!

File: app/townsquare/templates/townsquare/index.html (hunk 2 above)
Comment (zoek1) [reply app/townsquare/templates/townsquare/index.html]: ok, I'll remove this :smile:

File: app/assets/v2/js/status.js (hunk 1 above)
Comment (zoek1) [reply app/assets/v2/js/status.js]: Done!

File: app/townsquare/templates/townsquare/index.html (hunk 2 above)
Comment (zoek1) [reply app/townsquare/templates/townsquare/index.html]: Done! :raised_hands:

**Comment by zoek1** [comment]
The chages aré ready! 🙂

@octavioamu

**Pull request merged**